app/experimenter/nimbus-ui/src/components/FormBranches/index.test.tsx

```diff
@@ -219,31 +317,57 @@ describe("FormBranches", () => {
       description: "example description",
       ratio: "42",
       featureValue: "example value",
+      featureEnabled: true,
     };
 
     for (const id of [`branch-reference`, `branch-${branchIdx}`]) {
-      for (const [name, value] of Object.entries(expectedData)) {
-        const field = container.querySelector(`#${id}-${name}`);
-        expect(field).not.toBeNull();
-        fireEvent.change(field!, { target: { value } });
-      }
+      await fillInBranch(container, id, expectedData);
     }
 
     fireEvent.click(screen.getByTestId("save-button"));
     const saveResult = onSave.mock.calls[0][0];
-    expect(saveResult).toEqual({
-      featureConfig: MOCK_FEATURE_CONFIG_WITH_SCHEMA,
-      referenceBranch: {
-        ...MOCK_EXPERIMENT.referenceBranch,
-        ...expectedData,
-      },
-      treatmentBranches: [
-        MOCK_EXPERIMENT.treatmentBranches![0],
-        {
-          ...MOCK_EXPERIMENT.treatmentBranches![1],
-          ...expectedData,
-        },
-      ],
-    });
+
+    expect(saveResult.featureConfigId).toEqual(
+      parseInt(MOCK_FEATURE_CONFIG_WITH_SCHEMA.id, 10),
+    );
+    expect(saveResult.referenceBranch).toEqual(expectedData);
+    expect(saveResult.treatmentBranches[1]).toEqual(expectedData);
   });
 });
+
+async function fillInBranch(
+  container: HTMLElement,
+  id: string,
+  expectedData = {
+    name: "example name",
+    description: "example description",
+    ratio: "42",
+    featureValue: "example value",
+    featureEnabled: true,
+  },
+) {
+  for (const [name, value] of Object.entries(expectedData)) {
+    const field = container.querySelector(`#${id}-${name}`);
+    expect(field).not.toBeNull();
+    await act(async () => {
+      fireEvent.change(field!, { target: { value } });
+    });
+  }
+}
+
+function selectFeatureConfig(featureIdx = 1) {
+  const featureConfigAddButton = screen.queryAllByTestId(
+    "feature-config-add",
+  )![0];
```

Comment:
No, there are multiples of this and trying to query one appears to raise an error

Reply:
Ah, OK. I changed it and ran tests and thought things passed but I figured I might be missing something.

Reply:
Hmm, actually looking again, this *might* work. But only because there never happens to be multiple branches at the point where this gets used in tests. If/when there's a test that has multiple branches, each branch has a "+ Feature Config" button with data-testid="feature-config-add" and I think that throws an error